common/achievement/util.ts

```diff
@@ -67,19 +66,23 @@ export async function getBucketContext(userID: string, relation?: string): Promi
 
     logger.info('evaluate bucket configuration', { userType, relation, relationType, whereClause });
 
+    const nowPlusFiveMinutes = moment().add(5, 'minutes').toDate();
     let matches: any[] = [];
-    if (!relationType || relationType === 'match') {
+    if (relation?.includes('match') || !relation) {
         matches = await prisma.match.findMany({
             where: { ...whereClause, [`${userType}Id`]: id },
             select: {
                 id: true,
-                lecture: { where: { NOT: { declinedBy: { hasSome: [`${userType}/${id}`] } } }, select: { start: true, duration: true } },
+                lecture: {
+                    where: { start: { lte: nowPlusFiveMinutes }, NOT: { declinedBy: { hasSome: [`${userType}/${id}`] } } },
```

Comment:
filter to only get events, that already took place, guaranteeing our streaks to be evaluated with the correct dataset.